deps/chakrashim/src/v8trycatch.cc

```diff
@@ -78,23 +78,34 @@ void TryCatch::GetAndClearException() {
   }
 
   if (hasException) {
-    JsValueRef exceptionRef;
-    errorCode = JsGetAndClearException(&exceptionRef);
+    JsValueRef metadataRef;
+    errorCode = JsGetAndClearExceptionWithMetadata(&metadataRef);
     // We came here through JsHasException, so script shouldn't be in disabled
     // state.
     CHAKRA_ASSERT(errorCode != JsErrorInDisabledState);
     if (errorCode == JsNoError) {
-      error = exceptionRef;
+      metadata = metadataRef;
     }
   }
 }
 
 Handle<Value> TryCatch::ReThrow() {
-  if (error == JS_INVALID_REFERENCE) {
+  if (metadata == JS_INVALID_REFERENCE) {
     GetAndClearException();
   }
 
-  if (error == JS_INVALID_REFERENCE) {
+  if (metadata == JS_INVALID_REFERENCE) {
+    return Local<Value>();
+  }
```

Comment:
never mind, i see that `metadata` is updated inside `GetAndClearException()`. I misread it to be calling `JsGetAndClearException()`.

---
In reply to: [118545047](https://github.com/nodejs/node-chakracore/pull/263#discussion_r118545047) [](ancestors = 118545047)